File: 1904-second-largest-digit-in-a-string/second-largest-digit-in-a-string.cpp
class Solution {
public:
    int secondHighest(string s) {
        
         set<char>str;

         for(char ch : s){

              if(isdigit(ch)){

                   str.insert(ch-'0');
              }
         }
             if(str.size() < 2){

                  return -1;
             }

               auto it = str.end();
               it--;
               it--;

               return *it;
    }
};

File: 1904-second-largest-digit-in-a-string/second-largest-digit-in-a-string.java
class Solution {
    public int secondHighest(String s) {
        
         
          TreeSet <Integer> numbers = new TreeSet<>();

          for(char ch : s.toCharArray()){

               if(Character.isDigit(ch)){

                      numbers.add(ch - '0');
               }
          }
            if(numbers.size() < 2){

                  return -1;
            }

              Integer secondHighest = numbers.lower(numbers.last());

              return secondHighest != null ? secondHighest : -1;
    }
}